src/completion/indexProvider.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import { IArtifactSearchResult } from "../artifactSearcher";
+import { executeJavaLanguageServerCommand } from "../jdtls/commands";
+import { COMMAND_COMPLETION_ITEM_SELECTED, INFO_COMPLETION_ITEM_SELECTED } from "./constants";
+import { IMavenCompletionItemProvider } from "./IArtifactProvider";
+import { getSortText } from "./versionUtils";
+
+class IndexProvider implements IMavenCompletionItemProvider {
+    public async getGroupIdCandidates(groupIdHint: string, artifactIdHint: string): Promise<vscode.CompletionItem[]> {
+        const docs: IArtifactSearchResult[] = await executeJavaLanguageServerCommand("java.maven.searchArtifactByGA",
```

Comment:
how do you handle `JavaExtensionNotActivatedError` if redhat.java is not present?

Reply:
By use of global indicator as we have discussed.